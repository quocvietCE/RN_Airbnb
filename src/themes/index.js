import colors from './colors';
import Fonts from './fonts';
// import Metrics from './metrics';
import stylesApp from './stylesApp';

// export {colors, Fonts, Metrics, stylesApp};

export {colors, stylesApp, Fonts};
