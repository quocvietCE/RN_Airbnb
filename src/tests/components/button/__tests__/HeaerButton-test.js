import 'react-native';
import React from 'react';
import HeartButton from '../../../../components/buttons/HeartButton';
import {colors} from '../../../../themes';

import TestRenderer, {act} from 'react-test-renderer';

describe('Test HeartButton', () => {
  const mockOnPress = jest.fn();
  const componentTree = TestRenderer.create(
    <HeartButton
      color={colors.white}
      selectedColor={colors.pink}
      selected={true}
      onPress={mockOnPress}
    />,
  );

  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const componentSnapshot = componentTree.toJSON();
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  it('should call onPress', () => {
    const instance = componentTree.getInstance();
    instance.addToFavorite();
    // Act
    expect(mockOnPress).toHaveBeenCalled();
  });

  it('should call button testID="heartBtn"', () => {
    // press the button
    const button = componentTree.root.findByProps({testID: 'heartBtn'}).props;
    act(() => button.onPress());

    // expect test to equal "button pressed"
    // const
  });
});
