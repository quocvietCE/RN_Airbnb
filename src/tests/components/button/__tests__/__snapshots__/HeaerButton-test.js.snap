// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HeartButton Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="heartBtn"
>
  <View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 18,
          },
          Array [
            Object {
              "display": "none",
            },
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
